ts-client/src/dlmm/index.ts

```diff
@@ -2952,12 +2698,43 @@ export class DLMM {
     bps: BN;
     shouldClaimAndClose?: boolean;
   }): Promise<Transaction | Transaction[]> {
-    const { lbPair, lowerBinId, owner, feeOwner } =
-      await this.program.account.positionV2.fetch(position);
+    const lowerBinIdToRemove = Math.min(...binIds);
+    const upperBinIdToRemove = Math.max(...binIds);
+    const [lowerBinArrayPubKey] = deriveBinArray(this.pubkey, binIdToBinArrayIndex(new BN(binIds[0])), this.program.programId)
+    const [upperBinArrayPubKey] = deriveBinArray(this.pubkey, binIdToBinArrayIndex(new BN(binIds[binIds.length - 1])), this.program.programId)
+    const [positionAccInfo, lbPairAccInfo, upperBinArrayInfoAcc, lowerBinArrayInfoAcc] = await this.program.provider.connection.getMultipleAccountsInfo([
+      position,
+      this.pubkey,
+      upperBinArrayPubKey,
+      lowerBinArrayPubKey,
+    ])
+    const { lbPair, owner, feeOwner, upperBinId: positionUpperBinId, lowerBinId: positionLowerBinId } = this.program.coder.accounts.decode(
```

Comment:
No need to fetch for `lbPair`, `lower + upper` bin array to check if a position is empty. You just need `position` account and check `liquidityShares` are zero

Reply:
some state from lbPair still needed to pass to program interaction accounts param, removed lower + upper 👍

Reply:
This function is non-static, the `lbPair` state should already cached in the instance